ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1062,6 +1062,38 @@ void GCS_MAVLINK_Copter::handleMessage(mavlink_message_t* msg)
             break;
 #endif
 
+#if AC_RALLY == ENABLED
+        case MAV_CMD_NAV_RALLY_POINT: //not including home
+
+            if (!is_zero(packet.param1)) { // Go to a specific rally ID
+                copter.rally.cmd_go_to_rally_id = true;
+                copter.rally.go_to_rally_id = (uint8_t)packet.param2;
+            } else if (!is_zero(packet.param3)) { // Go to nearest rally from loc except home
+                copter.rally.cmd_go_to_rally_only = true;
+            } else if (!is_zero(packet.param4)) { // Go to new rally location (and save it in EEPROM)
+                RallyLocation rally_location;
+
+                rally_location.lat = (int32_t)(packet.param5 * 1.0e7f);
+                rally_location.lng = (int32_t)(packet.param6 * 1.0e7f);
+                rally_location.alt = (int32_t)(packet.param7);
+
+                // sanity check location
+                if (!check_latlng(rally_location.lat, rally_location.lng)) {
+                    break;
+                }
+
+                uint8_t new_rally_id = copter.rally.get_rally_point_total_count();
+                copter.rally.increment_rally_point_total_count();
+                copter.rally.set_rally_point_with_index(new_rally_id, rally_location);
+                copter.rally.cmd_go_to_rally_id = true;
+                copter.rally.go_to_rally_id = new_rally_id;
+            }
+
+            copter.set_mode(RTL, MODE_REASON_MISSION_END);
```

Comment:
This seems strange to me, it looks like Guided would be more appropriate, it's a request to go to a point, not an RTL.
Also, this certainly isn't reason mission end.

Reply:
I disagree, it's still a RTL for aborting / ending mission but with a special condition.
I can change to MODE_REASON_MISSION_COMMAND but I don't know where is the actual enum for this

Reply:
I'm confused, this isn't handled in a mission...

Reply:
Sorry, I meant MODE_REASON_GCS_COMMAND. (again I don't know the enums, just found these in the ArduRover set_mode reason..).
Still I think MISSION_END is not too far from the scope.

Reply:
Sorry, but I'm still confused, you said this is an RTL because it is for aborting/ending a mission, but this isn't handled as a command in a mission, this is handled as command coming from MAVLink, it doesn't matter what mode the vehicle is in.

Reply:
It's an outdated snippet of code, it was changed in the last update of this commit!

Reply:
It still does RTL which was my main point. Anyway, up to @rmackay9 to decide.